bme280 sample does not compile

**Describe the bug**
The https://docs.zephyrproject.org/latest/samples/sensor/bme280/README.html build instructions do not build. The recommended board (nrf52840_pca10056) doesn't have that sensor in its DTS, so the reason why is clear, but all in-tree samples should compile.

I am unable to find a board in tree which *does* have this sensor in its DTS.

It's not clear to me whether or how mainline CI is building this sample. It seems like the fixture restriction is bypassing it:

```
sanitycheck -vv -T samples/sensor/bme280/ -p nrf52840_pca10056
...
DEBUG   - nrf52840_pca10056         samples/sensor/bme280/sample.sensor.bme280         SKIPPED: No hardware support
DEBUG   - nrf52840_pca10056         samples/sensor/bme280/sample.sensor.bme280.spi     SKIPPED: No hardware support
```

**To Reproduce**
Steps to reproduce the behavior:
1. cd zephyr
2. west build -b nrf52840_pca10056 samples/sensor/bme280
3. observe build error

```
In file included from /home/mbolivar/zp/zephyr/drivers/sensor/bme280/bme280.c:24:0:
/home/mbolivar/zp/zephyr/drivers/sensor/bme280/bme280.h:114:2: error: #error "BME280 device type not specified"
 #error "BME280 device type not specified"
  ^~~~~
/home/mbolivar/zp/zephyr/drivers/sensor/bme280/bme280.c: In function 'bme280_init':
/home/mbolivar/zp/zephyr/drivers/sensor/bme280/bme280.c:370:22: warning: unused variable 'data' [-Wunused-variable]
  struct bme280_data *data = dev->driver_data;
                      ^~~~
In file included from ../include/drivers/sensor.h:23:0,
                 from /home/mbolivar/zp/zephyr/drivers/sensor/bme280/bme280.c:11:
/home/mbolivar/zp/zephyr/drivers/sensor/bme280/bme280.c: At top level:
/home/mbolivar/zp/zephyr/drivers/sensor/bme280/bme280.c:398:29: error: 'DT_INST_0_BOSCH_BME280_LABEL' undeclared here (not in a function); did you mean 'DT_INST_0_SOC_NV_FLASH_LABEL'?
 DEVICE_AND_API_INIT(bme280, DT_INST_0_BOSCH_BME280_LABEL, bme280_init, &bme280_data,
                             ^
../include/device.h:106:11: note: in definition of macro 'DEVICE_AND_API_INIT'
   .name = drv_name, .init = (init_fn),     \
           ^~~~~~~~
```

**Expected behavior**
Following build instructions for built-in samples should not result in compile errors.

**Environment (please complete the following information):**
 - Commit SHA or Version used: today's master (b96477dd69)

--- a/samples/sensor/bme280/nrf52840_pca10056.overlay
+++ b/samples/sensor/bme280/nrf52840_pca10056.overlay
@@ -1,0 +1,13 @@
+/*
+ * Copyright (c) 2020, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&i2c0 {
+	bme280@76 {
+		compatible = "bosch,bme280";
+		reg = <0x76>;
+		label = "BME280";
+	};
+};

--- a/samples/sensor/bme280/sample.yaml
+++ b/samples/sensor/bme280/sample.yaml
@@ -4,7 +4,7 @@ tests:
   sample.sensor.bme280:
     harness: console
     tags: sensors
-    depends_on: i2c bme280
+    platform_whitelist: nrf52840_pca10056
     harness_config:
         type: one_line
         regex:

--- a/samples/sensor/bme680/README.rst
+++ b/samples/sensor/bme680/README.rst
@@ -30,12 +30,12 @@ Building and Running
 This project outputs sensor data to the console. It requires a BME680
 sensor. It should work with any platform featuring a I2C peripheral interface.
 It does not work on QEMU.
-In this example below the :ref:`nrf52_pca10040` board is used.
+In this example below the :ref:`nrf52840_pca10056` board is used.
 
 
 .. zephyr-app-commands::
    :zephyr-app: samples/sensor/bme680
-   :board: nrf52_pca10040
+   :board: nrf52840_pca10056
    :goals: build flash
 
 Sample Output

--- a/samples/sensor/bme680/nrf52840_pca10056.overlay
+++ b/samples/sensor/bme680/nrf52840_pca10056.overlay
@@ -1,0 +1,13 @@
+/*
+ * Copyright (c) 2020, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&i2c0 {
+	bme680@76 {
+		compatible = "bosch,bme680";
+		reg = <0x76>;
+		label = "BME680";
+	};
+};

--- a/samples/sensor/bme680/sample.yaml
+++ b/samples/sensor/bme680/sample.yaml
@@ -4,4 +4,4 @@ tests:
   sample.sensor.bme680:
     harness: sensor
     tags: samples sensor
-    depends_on: i2c bme680
+    platform_whitelist: nrf52840_pca10056

